### PR TITLE
perf: Add SSR best benchmark

### PR DESCRIPTION
**Pull request by pmdartus**
## Details

This PR adds 2 new benchmarks to the existing benchmark suite: `benchmark-table/render/10k` and `benchmark-table-component/render/10k`. Those benchmarks measure the time it takes to create and render a list of 10k elements and a list of 10k components.

Along with `perf-benchmark` package has been restructured to contain both DOM and SSR performance benchmarks leveraging the same underlying LWC components.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅ 
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅ 

## GUS work item
W-7552299

### CODE REVIEW

**Comment by diervo** [comment]
@pmdartus Im wondering if you should create another project for SSR so we can run them in parallel.

File: packages/perf-benchmarks/src/__benchmarks__/engine-dom/benchmark-table-component/tablecmp-update-10th-1k.benchmark.js [hunk 1]
```diff
@@ -6,10 +6,10 @@
  */
 
 import { createElement } from 'lwc';
-import Table from 'benchmark/tableComponent';
 
-import Store from '../../tableStore';
-import { insertTableComponent, destroyTableComponent } from '../../utils';
+import Table from 'benchmark/tableComponent';
+import Store from 'benchmark/store';
+import { insertTableComponent, destroyTableComponent } from 'benchmark/utils';
```
Comment (ekashida) [inline packages/perf-benchmarks/src/__benchmarks__/engine-dom/benchmark-table-component/tablecmp-update-10th-1k.benchmark.js:12]: Looks like we need to either make the following change or rename `benchmark/utils/utils.js` to `benchmark/utils/index.js` and change the other module names to `benchmark/utils`?

```suggestion
import { insertTableComponent, destroyTableComponent } from 'benchmark/utils/utils';
```

**Comment by ekashida** [comment]
Are we missing a resolver for `benchmark/*`?

**Comment by diervo** [comment]
Didn’t have time to look into it, if piero doesn’t revolve it will give it
a run tomorrow

**Comment by pmdartus** [comment]
> Are we missing a resolver for benchmark/*?

@ekashida This is fixed now.

> However I still think we should have a separate CI job maybe so we can divide the time

@diervo The SSR tests are really fast right now (<10 seconds). Adding a new job on Circle CI will create more overhead than anything else. That being said, it would be great to leverage parallelization at the Best level instead of us creating multiple jobs.

**Comment by pmdartus** [comment]
This is probably due to the fact that new performance tests have been added between this branch and master. Best might not appreciate this.

**Pull request merged**